src/org/zaproxy/zap/extension/websocket/manualsend/WebSocketPanelSender.java

```diff
@@ -80,14 +94,95 @@ public void cleanup() {
     }
     
     private WebSocketProxy getDelegate(Integer channelId) {
-    	if (!connectedProxies.containsKey(channelId)) {
+    	if (closedProxies.containsKey(channelId)) {
     		logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
-    		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel") 
-    				+ " " + Constant.messages.getString("websocket.manual_send.fail"));
+    		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel")
+    				+ " " + Constant.messages.getString("websocket.manual_send.use_reopen"));
     	}
         return connectedProxies.get(channelId);
     }
 
+    /**
+     * Re-establish the connection
+	 *
+     * @param channelId connection to re-open
+     * @param newSecWebSocketKey New Sec-WebSocket-Key, null to use to generate new one
+	 * @param followRedirects true to follow redirections
+     * @param useTrackingSessionState useTrackingSessionState false for stateless http request
+	 * @throws IOException if connection refused
+	 * @throws RequestOutOfScopeException if url it's out of score ore ZAP runs to safe/protected mode
+	 * @return the new channel id
+     */
+    public int reOpenChannel(Integer channelId, String newSecWebSocketKey, boolean followRedirects, boolean useTrackingSessionState) throws IOException, RequestOutOfScopeException {
+    	WebSocketProxy connectionToEstablish = getWebSocketProxy(channelId);
+		HandshakeConfig handshakeConfig = connectionToEstablish.getHandShakeConfig(followRedirects, useTrackingSessionState);

```

Comment:
We need to handle the null config (for example, if the message was deleted), it's currently failing with a NPE:
```
Caused by: java.lang.NullPointerException
	at org.zaproxy.zap.extension.websocket.manualsend.WebSocketPanelSender.reOpenChannel(WebSocketPanelSender.java:123)
	at org.zaproxy.zap.extension.websocket.manualsend.ManualWebSocketSendEditorDialog$WebSocketMessagePanel$ReEstablishConnection.doInBackground(ManualWebSocketSendEditorDialog.java:436)
	at org.zaproxy.zap.extension.websocket.manualsend.ManualWebSocketSendEditorDialog$WebSocketMessagePanel$ReEstablishConnection.doInBackground(ManualWebSocketSendEditorDialog.java:422)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
That could be handled in two way. Either by building a new http handshake with the minimum requested headers or just inform the users that can establish a new WebSocket Connection (When the appropriate UI/API will be ready). I think the later way will be better because in some cases handshakes use additional header (ex Sec-WebSocket-Protocol). What do you think?

Reply:
Yeah, that sounds good to me.